src/notation/internal/notationpainting.cpp

```diff
@@ -60,6 +60,22 @@ void NotationPainting::setViewMode(const ViewMode& viewMode)
 
     m_viewModeChanged.notify();
     m_notation->notifyAboutNotationChanged();
+
+    if (viewMode != ViewMode::LINE && viewMode != ViewMode::HORIZONTAL_FIXED) {
+        return;
+    }
+
+    // VBoxes are not included in horizontal layouts - deselect them (and their contents) when switching to horizontal mode...
+    const std::vector<EngravingItem*> selection = score()->selection().elements();
```

Comment:
I feel like this is not the ideal place for this; it fits not ideally with the rest of this class. would it be an idea to do this in NotationInteraction, by subscribing to NotationPainting's `viewModeChanged` notification?

Reply:
Yeah you're absolutely right - putting it there also means no changes are required for the `INotationInteraction` interface (annoyed I didn't realise this myself!)